media_driver/agnostic/common/codec/hal/codechal_encoder_base.cpp

```diff
@@ -1209,26 +1209,29 @@ MOS_STATUS CodechalEncoderState::CheckResChangeAndCsc()
     {
         ResizeOnResChange();
     }
-#ifndef _FULL_OPEN_SOURCE
-    // check recon surface's alignment meet HW requirement
-    CODECHAL_ENCODE_CHK_STATUS_RETURN(m_cscDsState->CheckReconSurfaceAlignment(&m_reconSurface));
 
-    if (!m_cscDsState->IsEnabled() ||
-        CodecHal_PictureIsField(m_currOriginalPic) ||
-        CodecHal_PictureIsInterlacedFrame(m_currOriginalPic))
+    if (m_openCommonKernel)
```

Comment:
From programming perspective, should be, but actually we have this check earlier when create the encoder instance, if it's nullptr, it will failed at initialization phase.